File: lecture_010(RecursionwithArraylists)/l_011_printpermutations.java
import java.util.Scanner;
public class l_011_printpermutations {
    public static Scanner scn = new Scanner(System.in);
    public static void main(String[] args) {
        String str = scn.next();
        printPermutations(str,"");
    }
    public static void printPermutations(String ques, String ans) {
        if(ques.length()==0) {
            System.out.println(ans);
            return;
        }
        for(int i=0; i<ques.length(); i++) {
            char ch = ques.charAt(i);
            String left = ques.substring(0,i);
            String right = ques.substring(i+1);
            String roq = left + right;
            printPermutations(roq,ans+ch);
        }
    }
}

File: lecture_010(RecursionwithArraylists)/l_011_printpermutations.cpp
#include<iostream>
#include<string>
using namespace std;

void printpermutations(string ques, string ans) {
    if(ques.length()==0) {
        cout<<ans<<endl;
        return;
    }
    for(int i=0; i<ques.length(); i++) {
        char ch = ques[i];
        string left = ques.substr(0,i);
        string right = ques.substr(i+1);
        string roq = left + right;
        printpermutations(roq,ans+ch);
    }
}
int main() {
    string str;
    cin>>str;
    printpermutations(str,"");
}
